Integrations/GitHub/GitHub.py

```diff
@@ -0,0 +1,277 @@
+            print_results['HumanReadable'] = 'Issue #' + str(issue_num) + ' has been closed'
+            print_results['EntryContext'] = {
+                'GitHub.Issues(val.ID==obj.ID)': {'ID': str(r['id']), 'state': 'close'}
+            }
+        else:
+            print_results['HumanReadable'] = 'No such issue'
+        print_results['Contents'] = r
+        demisto.results(print_results)
+    else:
+        demisto.results({
+            'Type': entryTypes['note'],
+            'ContentsFormat': formats['text'],
+            'Contents': r,
+            'HumanReadable': 'Please enter a valid issue number',
+        })
+    return r
+
+
+def make_search(search_query, sort, order):
+    query = '?q=repo:teizenman/Demisto-start'
+    if search_query:
+        query += '+' + search_query
+    if sort:
+        query += '&sort=' + sort
+        if order:
+            query += '&order=' + order
+    r = http_request('GET', 'search/issues' + query)
+    print_results = {
+        'Type': entryTypes['note'],
+        'ContentsFormat': formats['markdown'],
+        'Contents': r
+    }
+    if r['total_count'] != 0:
+        results = []
+        for issue in r['items']:
+            issue_details = makehash()
+            issue_details['ID'] = str(issue['id'])
+            issue_details['title'] = issue['title']
+            issue_details['link'] = issue['url']
+            issue_details['state'] = issue['state']
+            results.append(issue_details)
+            print_results['HumanReadable'] = tableToMarkdown(
+                str(r['total_count']) + ' found issues.\nComplete list of issues: ' + str(not r['incomplete_results']),
+                results, ['ID', 'title', 'state', 'link'])
+        print_results['EntryContext'] = {
+            'GitHub.Searches(val.query==obj.query)': {'query': query, 'results': results}
+        }
+    else:
+        print_results['HumanReadable'] = 'No issues found for this query!'
+    demisto.results(print_results)
+    return r
+
+
+def count_total_downloads():
+    r = http_request('GET', 'repos/teizenman/Demisto-start/releases')
+    output = []
+    for asset in r[0]['assets']:
+        count = makehash()
+        count['asset_name'] = asset['name']
+        count['download_count'] = asset['download_count']
+        output.append(count)
+    demisto.results({
+        'Type': entryTypes['note'],
+        'ContentsFormat': formats['markdown'],
+        'Contents': r,
+        'HumanReadable': tableToMarkdown('Download counts:', output, ['asset_name', 'download_count']),
+        'EntryContext': {
+            'GitHub.Download-counts(val.asset_name==obj.asset_name)': output
+        }
+    })
+    return r
+
+
+''' EXECUTION '''
+LOG('command is %s' % (demisto.command(),))
+try:
+    if demisto.command() == 'list-all-issues':
+        list_all_issues()
+
+    elif demisto.command() == 'create-issue':
+        title = demisto.args().get('title')
```

Comment:
we generally split this into two functions or more, one should be the one doing the business logic, and the other one should be handling the request to the service.
All the demisto.args() parts should be in the business logic which calls the request function